### PR TITLE
Handle end of life Debian versions

### LINKED ISSUE #564: Debian version exception
```
 File "/usr/local/bin/worker.py", line 531, in _do_process_task
    self._source_update(message)
  File "/usr/local/bin/worker.py", line 381, in _source_update
    self._do_update(source_repo, repo, vulnerability, path, original_sha256)
  File "/usr/local/bin/worker.py", line 458, in _do_update
    result = self._analyze_vulnerability(source_repo, repo, vulnerability,
  File "/usr/local/bin/worker.py", line 426, in _analyze_vulnerability
    result = osv.analyze(
  File "/env/osv/impact.py", line 492, in analyze
    enumerate_versions(affected_package.package.name,
  File "/env/osv/impact.py", line 400, in enumerate_versions
    current_versions = ecosystem.enumerate_versions(
  File "/env/osv/ecosystems.py", line 353, in enumerate_versions
    versions.sort(key=self.sort_key)
  File "/env/osv/ecosystems.py", line 336, in sort_key
    return DebianVersion.from_string(version)
  File "/env/osv/third_party/univers/debian.py", line 141, in from_string
    raise ValueError('Invalid version string: "{}"'.format(version))
ValueError: Invalid version string: "1:Beta2-2.1"
```

### PR DESCRIPTION
**Pull request by another-rex**
Fixes #564

### CODE REVIEW

**Review by oliverchang — COMMENTED** [review]
Thanks! 

Did you also get a chance to check out the exception in https://github.com/google/osv.dev/issues/564#issue-1322858420 ?

**Comment by another-rex** [comment]
Oh my bad, thought they were the same exception. This only fixes the one in the comment, having a look at the first one now.

**Comment by oliverchang** [comment]
> Oh my bad, thought they were the same exception. This only fixes the one in the comment, having a look at the first one now.

No worries :) Was just asking since this PR will close the linked issue. 

Otherwise this PR itself LGTM on its own (assuming we remove that link).

**Review by oliverchang — APPROVED** [review]

**Comment by another-rex** [comment]
Actually fixes the original issue now

File: osv/ecosystems.py [hunk 1]
```diff
@@ -349,10 +355,19 @@ def enumerate_versions(self, package, introduced, fixed, limits=None):
       ) from ex
 
     response = json.loads(text_response)
-    versions: list[str] = [x['version'] for x in response['result']]
+    raw_versions: list[str] = [x['version'] for x in response['result']]
+
+    # Remove rare cases of unknown versions
+    def version_is_valid_log(v):
```
Comment (oliverchang) [inline osv/ecosystems.py:361]: nit: no need for `_log`. It's a bit confusing what the return value means otherwise.

**Review by oliverchang — APPROVED** [review]
thanks!

**Pull request merged**